iocore/net/SSLNetVConnection.cc

```diff
@@ -207,11 +207,17 @@ ssl_read_from_net(SSLNetVConnection *sslvc, EThread *lthread, int64_t &ret)
   while (sslErr == SSL_ERROR_NONE) {
     int64_t block_write_avail = buf.writer()->block_write_avail();
     if (block_write_avail <= 0) {
-      buf.writer()->add_block();
-      block_write_avail = buf.writer()->block_write_avail();
-      if (block_write_avail <= 0) {
-        Warning("Cannot add new block");
+      // If we filled up one block, give back to the event loop so we don't
+      // overbuffer.
+      if (bytes_read > 0) {
```

Comment:
I believe this condition check breaks the watermark design in IOBuffer.

We can use `readv()` in `UnixNetVC::read_from_net` to read data into multi IOBufferBlocks if the rest space in the first IOBufferBlock is less than watermark size.

But there is only `SSLReadBuffer`, ATS should `read()` data for every IOBufferBlocks that is attached into MIOBuffer.